crates/driver/src/domain/competition/solution/encoding.rs

```diff
@@ -203,48 +206,45 @@ pub fn tx(
         interactions.push(unwrap(native_unwrap, contracts.weth()));
     }
 
-    let tx = contracts
+    let interactions = [
+        pre_interactions.iter().map(codec::interaction).collect(),
+        interactions.iter().map(codec::interaction).collect(),
+        post_interactions.iter().map(codec::interaction).collect(),
+    ];
+
+    let has_flashloans = !solution.flashloans.is_empty();
+    let has_wrappers = !solution.wrappers.is_empty();
+
+    // Check for incompatibility upfront
+    if has_flashloans && has_wrappers {
+        return Err(Error::FlashloanWrappersIncompatible);
+    }
```

Comment:
nit: The logic to encode the settlement call data is not expensive so I'd opt for always encoding the `settle()` calldata and moving this error handling to the `if-else` branches further down to keep everything close together.